Download subset fix data with python script

weihuang-jedi opened the pull request:
Using python script to download a subset of FIX data from NOAA s3 bucket for a specific (low res) grid to a local machine. This is usually when user try to setup Global Workflow on a new platform, where NOAA EMC GW FIX data is not available yet.

  Refs #3254

This PR right now serve as the test PR to activate CI testing on AWS.

Please do not merge it. And later we need Rahul to review and decide to put it at proper directory.

# Type of change
- [ ] Bug fix (fixes something broken)
- [x ] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
<!-- Choose YES or NO from each of the following and delete the other -->
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO (If YES, please add a link to any PRs that are pending.)
  - [ ] EMC verif-global <!-- NOAA-EMC/EMC_verif-global#1234 -->
  - [ ] GDAS <!-- NOAA-EMC/GDASApp#1234 -->
  - [ ] GFS-utils <!-- NOAA-EMC/gfs-utils#1234 -->
  - [ ] GSI <!-- NOAA-EMC/GSI#1234 -->
  - [ ] GSI-monitor <!-- NOAA-EMC/GSI-Monitor#1234 -->
  - [ ] GSI-utils <!-- NOAA-EMC/GSI-Utils#1234 -->
  - [ ] UFS-utils <!-- ufs-community/UFS_UTILS#1234 -->
  - [ ] UFS-weather-model <!-- ufs-community/ufs-weather-model#1234 -->
  - [ ] wxflow <!-- NOAA-EMC/wxflow#1234 -->

# How has this been tested?
Tested on AWS.

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

weihuang-jedi committed:
change noaacloud NodeName

weihuang-jedi committed:
Merge branch 'develop' of github.com:NOAA-EPIC/global-workflow-cloud into develop

weihuang-jedi committed:
add download fix subset data, also serve as a test PR to trigger CI testing on AWS

weihuang-jedi committed:
fix pynorm error

weihuang-jedi committed:
fix pynorm error 2

weihuang-jedi committed:
fix pynorm error3

weihuang-jedi committed:
fix pynorm error 4

weihuang-jedi committed:
fix pynorm error 5

weihuang-jedi committed:
fix pynorm error 6

weihuang-jedi committed:
fix pynorm error 7

weihuang-jedi committed:
fix pynorm error 8

weihuang-jedi committed:
fix pynorm error 9

weihuang-jedi committed:
fix pynorm error 10

weihuang-jedi commented:
This PR right now serve as the test PR to activate CI testing on AWS.

Please do not merge it. And later we need Rahul to review and decide to put it at proper directory.

weihuang-jedi committed:
fix a syntax error

weihuang-jedi committed:
use just /lutre/jenkins for CI testing work directory

weihuang-jedi committed:
Merge branch 'develop' of github.com:NOAA-EPIC/global-workflow-cloud into download-subset-fix-data

weihuang-jedi committed:
using argparse and logging instead of getopt and print

weihuang-jedi committed:
using argparse and logging instead of getopt and print

aerorahul commented on ush/fetch-fix-data.py:
```diff
@@ -0,0 +1,334 @@
+# -----------------------------------------------------------------------------
+
+
+def namespace_to_dict(namespace):
+    return {
+        k: namespace_to_dict(v) if isinstance(v, argparse.Namespace) else v
+        for k, v in vars(namespace).items()
+    }
```
you can just do `return vars(namespace)`

weihuang-jedi replied:
This not need any more.

kbooker79 committed:
Updating with proper nomenclature.

kbooker79 committed:
Merge pull request #5 from NOAA-EPIC/download-subset-fix-data

Download subset fix data

kbooker79 committed:
Updating for CI.

kbooker79 committed:
Updating.

kbooker79 committed:
Updating.

kbooker79 committed:
Fixing syntax error.

weihuang-jedi committed:
consist with Terry's code

weihuang-jedi committed:
add a ls command to make sure code in cloned

weihuang-jedi committed:
try clone the code directly

weihuang-jedi committed:
use https to clone

weihuang-jedi committed:
Merge branch 'NOAA-EMC:develop' into develop

weihuang-jedi committed:
add more debug ls

weihuang-jedi committed:
adding comments

weihuang-jedi committed:
fix pynorm error

weihuang-jedi committed:
add more comments

weihuang-jedi committed:
add more comments, fixed a quota issue

weihuang-jedi committed:
sync

weihuang-jedi committed:
fix a typo

weihuang-jedi committed:
remove 2 comments

weihuang-jedi committed:
remove conflict

weihuang-jedi committed:
reset HOMEgfs

weihuang-jedi committed:
compile for gfs only for now

weihuang-jedi committed:
comment gh pr eidt for now

weihuang-jedi committed:
skip CI on AWS

weihuang-jedi committed:
Merge remote-tracking branch 'origin/develop' into download-subset-fix-data

weihuang-jedi committed:
trying to fix runtime bug

weihuang-jedi committed:
still trying to figure out HOEgfs issue

weihuang-jedi committed:
add compile gefs

weihuang-jedi committed:
add compile gefs

weihuang-jedi committed:
add compile gefs

weihuang-jedi committed:
add main function

weihuang-jedi committed:
switch back to 'checkout scm instead of git clone'

weihuang-jedi committed:
using checkout scm

weihuang-jedi committed:
using checkout scm

kbooker79 committed:
Updating to use SCM.

weihuang-jedi committed:
remove memory from resource if on AWS

weihuang-jedi committed:
remove memory requirement for AWS

weihuang-jedi committed:
Merge branch 'NOAA-EMC:develop' into develop

weihuang-jedi committed:
sync

weihuang-jedi committed:
fix pynorms error

weihuang-jedi committed:
sync

weihuang-jedi committed:
Update C48_S2SW.yaml

weihuang-jedi commented:
Please dot review yet, please do not merge.
Use this one as test case for Enabling CI Testing on AWS.
Thanks,
Wei

weihuang-jedi committed:
sync

weihuang-jedi committed:
sync

weihuang-jedi committed:
Merge branch 'develop' into download-subset-fix-data

weihuang-jedi committed:
Merge branch 'develop' into download-subset-fix-data

weihuang-jedi commented:
@aerorahul Could you please review this PR?
I have made some changes after our chat on this code,
please review again to see if there is more to modify.
Thanks,
Wei

weihuang-jedi committed:
Merge remote-tracking branch 'origin/develop' into download-subset-fix-data

aerorahul commented on ush/fetch-fix-data.py:
```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python
```
```suggestion
#!/usr/bin/env python3
```

aerorahul commented on ush/fetch-fix-data.py:
```diff
@@ -0,0 +1,330 @@
+# 2025-02-26
+# script to download a subset of FIX data to local machines.
+import os
+import time
+import sys
+import getopt
```
```suggestion
```
Not used

aerorahul commented on ush/fetch-fix-data.py:
```diff
@@ -0,0 +1,330 @@
+    args = parser.parse_args()
+
+    if args.verbose:
+        print(f"the atmgrid is {args.atmgrid}")
+        print(f"the ocngrid is {args.ocngrid}")
+        print(f"the localdir is {args.localdir}")
+        print(f"the fix_file is {args.fix_ver}")
+        print(f"the s3 bucket is {args.fix_bucket}")
```
Use logging

weihuang-jedi replied:
Yes. will make changes to use logging.

aerorahul commented on ush/fetch-fix-data.py:
```diff
@@ -0,0 +1,330 @@
+        ----------
+        None
+        """
+        for atmgrid in self.atmgridarray:
+            for ocngrid in self.ocngridarray:
+                newkey = f'{key}_a{atmgrid}o{ocngrid}'
+                self.s3dict[newkey] = f'{key}/{val}/a{atmgrid}o{ocngrid}'
```
I don't think we have every ocean resolution available for an atmosphere grid, but OK!

weihuang-jedi replied:
That is true. Like low res. ATM won't have high res. OCN, and hi res. ATM no point have low res. OCN.
Will try to improve later.

aerorahul commented:
I ran this branch for a test on C48 and mx500.  It downloaded the files.
It also created a logfile in `ush/fetch-fix-data.log`. This file should be added to the `.gitignore` list as it makes the git clone dirty.
I then re-executed the script to download C96 and mx500.  It downloaded the C96 relevant files.

The script could use more work to follow the guidelines of this project, but we can postpone that for another time.

Please update `.gitignore` and apply recommended review suggestions and we can merge this PR.

weihuang-jedi committed:
Update ush/fetch-fix-data.py

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

weihuang-jedi committed:
Update ush/fetch-fix-data.py

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

weihuang-jedi committed:
Merge branch 'download-subset-fix-data' of github.com:NOAA-EPIC/global-workflow-cloud into download-subset-fix-data

weihuang-jedi committed:
ignore fetch-fix-data.log, remove unused module from script, and switch more print to logger.info

weihuang-jedi committed:
Merge branch 'develop' into download-subset-fix-data

weihuang-jedi commented:
@aerorahul I have updated the suggested changes, added ush/fetch-fix-data.log to .gitignore file.
Please review again, and see if I missed anything.
Thanks,
Wei
